qa/rpc-tests/prioritise_transaction.py

```diff
@@ -0,0 +1,147 @@
+
+    def create_confirmed_utxos(self, count):
+        self.nodes[0].generate(int(0.5*count)+101)
+        utxos = self.nodes[0].listunspent()
+        iterations = count - len(utxos)
+        addr1 = self.nodes[0].getnewaddress()
+        addr2 = self.nodes[0].getnewaddress()
+        if iterations <= 0:
+            return utxos
+        for i in xrange(iterations):
+            t = utxos.pop()
+            fee = self.relayfee
+            inputs = []
+            inputs.append({ "txid" : t["txid"], "vout" : t["vout"]})
+            outputs = {}
+            send_value = t['amount'] - fee
+            outputs[addr1] = satoshi_round(send_value/2)
+            outputs[addr2] = satoshi_round(send_value/2)
+            raw_tx = self.nodes[0].createrawtransaction(inputs, outputs)
+            signed_tx = self.nodes[0].signrawtransaction(raw_tx)["hex"]
+            txid = self.nodes[0].sendrawtransaction(signed_tx)
+
+        while (self.nodes[0].getmempoolinfo()['size'] > 0):
+            self.nodes[0].generate(1)
+
+        utxos = self.nodes[0].listunspent()
+        assert(len(utxos) >= count)
+        return utxos
+
+    def create_lots_of_big_transactions(self, utxos, fee):
+        addr = self.nodes[0].getnewaddress()
+        txids = []
+        for i in xrange(len(utxos)):
+            t = utxos.pop()
+            inputs = []
+            inputs.append({ "txid" : t["txid"], "vout" : t["vout"]})
+            outputs = {}
+            send_value = t['amount'] - fee
+            outputs[addr] = satoshi_round(send_value)
+            rawtx = self.nodes[0].createrawtransaction(inputs, outputs)
+            newtx = rawtx[0:92]
+            newtx = newtx + self.txouts
+            newtx = newtx + rawtx[94:]
+            signresult = self.nodes[0].signrawtransaction(newtx, None, None, "NONE")
+            txid = self.nodes[0].sendrawtransaction(signresult["hex"], True)
+            txids.append(txid)
+        return txids
+
+    def run_test(self):
+        utxos = self.create_confirmed_utxos(90)
+        base_fee = self.relayfee*100 # our transactions are smaller than 100kb
+        txids = []
+
+        # Create 3 batches of transactions at 3 different fee rate levels
+        for i in xrange(3):
+            txids.append([])
+            txids[i] = self.create_lots_of_big_transactions(utxos[30*i:30*i+30], (i+1)*base_fee)
+
+        # add a fee delta to something in the cheapest bucket and make sure it gets mined
+        # also check that a different entry in the cheapest bucket is NOT mined (lower
+        # the priority to ensure its not mined due to priority)
+        self.nodes[0].prioritisetransaction(txids[0][0], 0, int(3*base_fee*COIN))
+        self.nodes[0].prioritisetransaction(txids[0][1], -1e15, 0)
+
+        self.nodes[0].generate(1)
+
+        mempool = self.nodes[0].getrawmempool()
+        assert(txids[0][0] not in mempool)
+        assert(txids[0][1] in mempool)
+        print "Prioritised transaction mined: success"
```

Comment:
Well then it would be printing something incorrect if the assertion failed.  I guess I could add a different print statement above the asserts, but not sure it's worth it?  If the assertion fails I think the stack trace is informative enough, and I don't think the style here is inconsistent with the rest of the tests.